src/compiler/core.ts

```diff
@@ -1394,14 +1394,28 @@ namespace ts {
         }
     }
 
-    export function copyListRemovingItem<T>(item: T, list: T[]) {
-        const copiedList: T[] = [];
-        for (const e of list) {
-            if (e !== item) {
-                copiedList.push(e);
+    /** Remove an item from an array, moving everything to its right one space left. */
+    export function removeItemAt<T>(array: T[], index: number): void {

```

Comment:
Just a thought, if in most cases the array does not care about order, you can just set the remove position item to the last one, and then call `pop` and done:
```
array[index] = array[array.length - 1];
array.pop();
```
which should be a lot faster. Maybe it can be called something like `unOrderedRemoveItemAt`